### PR TITLE
Fix non-screen/component Prettier/ESLint changes

### PR DESCRIPTION
**Pull request by david-qing**
### :cloud: :cloud: Documentation :cloud: :cloud:

_Describe your feature & code in detail, so that future developers know how to handle it_

* Made Prettier/ESLint changes for non-screen and non-component files. Applied best practices to these files, ensuring that code is clean. 
* Fixed small bug in RealmApp.tsx and sync.config.js in which `mongodb = mongodb = ""` was typed instead of `mongodb = ""`.  
* Fixed issues with fonts by uncommenting font edits, and running `npx react-native-asset`. This change linked the fonts to the build, and loads the fonts each time the application is built. 
* For ESLint errors that were intended, added suppression to make sure the errors do not show up in logs. 

### :zap: Testing :zap:

_List all manual tests done/new unit tests you have implemented_

Tested by running the app: `npx expo run:ios`. Application fully loads, and navigation to all screens works properly with no crashes or errors. 

### Screenshots :calling:

_Required for visual changes - please CC Esther (leexesther) below if adding screenshots!_

## :seedling: Related PRs :seedling:

_Describe any other PRs that might affect/depend on yours_

#11 Disable React import
Change in PR #11 ensures that some changes in this PR are applied properly. 

#8 Adding Prettier/ESLint
Change in this PR added Prettier/ESLint. 

**_Optional - any related PRs you're waiting on, or PRs that will conflict, etc_**

#11 Might have merge conflicts, easy to fix. 

### :space_invader: Any notable bugs you encountered or are still having trouble with? :space_invader:

Ensure that running the app on different computers still loads properly; some changes could be bug-ridden. 

### :boom: TODO:

Implement UserStack and AuthStack switching for navigation purposes. 

Make the decision on how i18n should be implemented; one file per language or one big file containing all translations?

CC: @davidqing6432
@wangannie

### CODE REVIEW

File: metro.config.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
 // Learn more https://docs.expo.io/guides/customizing-metro
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
 const { getDefaultConfig } = require('expo/metro-config');
+// TODO: Implement using an import statement instead of require. 
+// import { getDefaultConfig } from 'expo/metro-config';
```
Comment (wangannie) [inline metro.config.js]: You can remove this. I think this is fine to keep the `require` and use the eslint override (at least that's what I did for vouchers)

File: metro.config.js [hunk 2]
```diff
@@ -1,4 +1,10 @@
...
 
 module.exports = getDefaultConfig(__dirname);
+// TODO: Implement using export statement instead of module.exports
```
Comment (wangannie) [inline metro.config.js]: Why?

File: providers/AuthProvider.tsx [hunk 3]
```diff
@@ -73,37 +75,33 @@ const AuthProvider = ({ children }: any) => {
...
-  // logged in user
-  const signOut = () => {
-    if (user == null) {
-      console.warn("Not logged in, can't log out!");
-      return;
-    }
-    user.logOut();
-    setUser(null);
-  };
+  const authFns = useMemo(() => {
```
Comment (wangannie) [inline providers/AuthProvider.tsx:77]: I didn't test functionality (bc I can't install XCode) for this bc I'm assuming you did & it works? If so, lgtm

File: src/i18n/i18n.constants.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
-export const namespaces = { //how we break translations up: we could do it all on one page or seperate common ones
+// TODO: Decide whether to split up translations into mutiple common files, or do all in a single file.
```
Comment (wangannie) [inline src/i18n/i18n.constants.ts]: Nit: try to always tag someone on your TODO comments so you remember who should address them later!

e.g. 

```
// TODO @david: Decide....
```

File: src/i18n/i18n.ts [hunk 5]
```diff
@@ -18,4 +18,5 @@ const createI18n = (language: string): i18nInstance => {
   return i18n;
 };
 
-export const i18n = createI18n(languages.es);
\ No newline at end of file
+const i18n = createI18n(languages.es);
+export default i18n;
```
Comment (wangannie) [inline src/i18n/i18n.ts]: Make sure to run prettier on this!

If you don't have prettier running on save, you can run the autofix on a specific file/directory using this command:
```
npx prettier --write {filepath i.e. /Users/anniewang/Development/test/cultural-survival-mobile/src/i18n/i18n.constants.ts}
```

File: src/models/Task.ts [hunk 6]
```diff
@@ -29,3 +34,5 @@ export class Task extends Realm.Object {
     },
   };
 }
+
+export default Task;
```
Comment (wangannie) [inline src/models/Task.ts]: Run prettier -- see prev comment

File: src/models/index.ts [hunk 7]
```diff
@@ -1,6 +1,8 @@
 import {createRealmContext} from '@realm/react';
-import {Task} from './Task';
+import Task from './Task';
 
-export const TaskRealmContext = createRealmContext({
+const TaskRealmContext = createRealmContext({
```
Comment (wangannie) [inline src/models/index.ts:4]: When is this used? Make sure the sites that use this are updated.

File: src/navigation/userStack.tsx [hunk 8]
```diff
@@ -1,24 +1,25 @@
 // import { createNativeStackNavigator } from "@react-navigation/native-stack";
```
Comment (wangannie) [inline src/navigation/userStack.tsx]: Remove this

File: src/navigation/userStack.tsx [hunk 9]
```diff
@@ -1,24 +1,25 @@
...
-import { NavigationContainer } from "@react-navigation/native";
-import { createStackNavigator } from "@react-navigation/stack";
+import React from 'react';
```
Comment (wangannie) [inline src/navigation/userStack.tsx]: Rebase onto main (since the #11 change was merged) and this should be removed!

File: src/styles/button.ts [hunk 10]
```diff
@@ -1,7 +1,7 @@
 import {StyleSheet} from 'react-native';
 import colors from './colors';
 
-export const buttonStyles: StyleSheet.NamedStyles<any> = {
+const buttonStyles = StyleSheet.create({
```
Comment (wangannie) [inline src/styles/button.ts:4]: Here (and elsewhere) feel free to use `export default` directly!

e.g. `export default const buttonStyles...`

Not a huge deal either way, just a shortcut to use less lines.

File: src/styles/button.ts [hunk 11]
```diff
@@ -17,4 +17,6 @@ export const buttonStyles: StyleSheet.NamedStyles<any> = {
     fontSize: 17,
     fontWeight: 'bold',
   },
-};
+});
+
+export default buttonStyles;
```
Comment (wangannie) [inline src/styles/button.ts]: Run prettier!

**Review by wangannie — CHANGES_REQUESTED** [review]
Linter fixes look really good! Good job with this 😄 Since I can't get XCode installed, I wasn't able to test functionality, but I trust that you took care of that.

A few things:
- Rebase onto the latest main (since #11 got merged) and remove React imports
- It doesn't seem like you ran prettier on all files (based on some spacing discrepancies and missing newlines at the ends of files?), so make sure you do that on the files you're changing. If you have format on save enabled ([see this guide on how to do that](https://scottsauber.com/2017/06/10/prettier-format-on-save-never-worry-about-formatting-javascript-again/)), it should prettify automatically! Once you prettify, I'm good to approve. 
  - To check that there are no prettier errors, run `npm run prettier:check` and make sure the files you're fixing aren't included in the error output!
- Nit on your PR description: it's helpful to break your changes into bullet points!

File: providers/AuthProvider.tsx (hunk 3 above)
Comment (david-qing) [reply providers/AuthProvider.tsx:77]: to my knowledge, this worked properly! will test one more time

Comment (david-qing) [reply providers/AuthProvider.tsx:77]: using the hardcoded test login button, it seems to access auth properly!

File: src/models/index.ts (hunk 7 above)
Comment (david-qing) [reply src/models/index.ts:4]: not used anywhere, is purely example code!

File: src/styles/button.ts (hunk 10 above)
Comment (david-qing) [reply src/styles/button.ts:4]: `export default const` was not working for me, so i am just leaving it in this two line format!

File: src/navigation/authStack.tsx [hunk 12]
```diff
@@ -1,12 +1,38 @@
-import { NavigationContainer } from "@react-navigation/native";
-import { createStackNavigator } from "@react-navigation/stack";
+import React from 'react';
```
Comment (wangannie) [inline src/navigation/authStack.tsx]: Changes in this screen should probably be in a separate PR, since they're completely unrelated.

File: src/navigation/index.tsx [hunk 13]
```diff
@@ -1,11 +1,12 @@
 // import { useAuthentication } from '../utils/hooks/useAuthentication';
```
Comment (wangannie) [inline src/navigation/index.tsx]: Remove this comment

File: src/screens/SignIn1Screen/SignIn1.tsx [hunk 14]
```diff
@@ -36,30 +36,30 @@ const SignIn1Screen = ({ navigation }: any) => {
             Advancing Indigenous Peoples' Rights & Cultures Worldwide.
           </Text>
         </View>
-        {/* TODO: Potentially extract styles into the button component */}
+        {/* TODO @auth-flow: Potentially extract styles into the button component */}
```
Comment (wangannie) [inline src/screens/SignIn1Screen/SignIn1.tsx:39]: I like this @ auth-flow! good thinking : ) makes it much easier to search and resolve all the auth TODOs later

File: src/types/navigation.tsx [hunk 15]
```diff
@@ -7,6 +7,7 @@ import { NativeStackScreenProps } from '@react-navigation/native-stack';
```
Comment (wangannie) [inline src/types/navigation.tsx:6]: Not related to your changes, but I noticed there's an eslint-disable-next-line on line 5 of this file that I'm pretty sure doesn't need to be there, since the package is indeed imported in package.json

**Review by wangannie — APPROVED** [review]
Looks great! Most comments are nits. One thing you should look at before merging is the changes in `AuthStack` -- not sure if you intended to include them in this PR?
